src/frequenz/lib/notebooks/config.py

```
@@ -126,22 +129,45 @@ def component_types(self) -> list[str]:
         """Get a list of all component types in the configuration."""
         return list(self._component_types_cfg.keys())
 
-    def component_type_ids(self, component_type: str) -> list[int]:
+    def component_type_ids(
+        self, component_type: str, component_category: str | None = None
+    ) -> list[int]:
         """Get a list of all component IDs for a component type.
 
         Args:
             component_type: Component type to be aggregated.
+            component_category: Specific category of component IDs to retrieve
+                (e.g., "meter", "inverter", or "component"). If not provided,
+                the default logic is used.
 
         Returns:
             List of component IDs for this component type.
 
         Raises:
             ValueError: If the component type is unknown.
+            KeyError: If `component_category` is invalid.
+            LookupError: If the `component_category` is valid but not available.
         """
         cfg = self._component_types_cfg.get(component_type)
         if not cfg:
             raise ValueError(f"{component_type} not found in config.")
 
+        if component_category:
+            valid_categories = get_args(ComponentCategory)
+            if component_category not in valid_categories:
+                raise KeyError(
+                    f"Invalid component category: {component_category}. "
+                    f"Valid categories are {valid_categories}"
+                )
+            category_ids = cast(
+                list[int] | None, getattr(cfg, component_category, None)
+            )
+            if category_ids:
+                return category_ids
+            raise LookupError(
```

I am not sure if we should raise here. E.g. having a microgrid without an inverter is totally fine, whether this is a problem or not depends on the use-case.

---

You’re right that having a microgrid without certain component categories can be valid depending on the use case. My initial reasoning for raising an exception here was to ensure that missing IDs for a specified category would be flagged explicitly, helping to catch potential configuration issues early and avoid silent failures. However, I understand that this strict validation may not align with all use cases. Returning an empty list instead would provide more flexibility, allowing downstream code to handle the absence of components as needed without assuming it's an error.